src/Core/Domain/Country/Query/GetCountries.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Country\Query;
+
+/**
+ * Class is responsible for providing countries.
+ */
+class GetCountries
+{
+    /**
+     * @var int
+     */
+    private $langId;
+
+    /**
+     * @var bool
+     */
+    private $active = false;
+
+    /**
+     * @var bool
+     */
+    private $containsStates = false;
+
+    /**
+     * @var bool
+     */
+    private $listStates = false;
+
+    /**
+     * @param int $langId
+     */
+    public function __construct(int $langId)
+    {
+        $this->langId = $langId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getLangId(): int
+    {
+        return $this->langId;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isActive(): bool
+    {
+        return $this->active;
+    }
+
+    /**
+     * @param bool $active
+     *
+     * @return GetCountries
+     */
+    public function setActive(bool $active): GetCountries
```

Comment:
What if we drop setters here and instead use something like this:
```php
$query = (new GetCountries($langId))
    ->onlyActive()
    ->thatContainsStates()
;
```

@matks wdyt?

Reply:
Well get countries options are passed from form configuration so i think it is more clear to just pass those options to setters instead of writing additional conditions to call methods